File: java_problems/src/Ages.java
import java.util.Scanner;
import java.util.Locale;

public class Ages {
    public static void main(String[] args) {
        // instances
        Scanner sc = new Scanner(System.in);
        Locale.setDefault(Locale.US);
        
        // variables
        String name1, name2;
        int age1, age2;
        double average;

        // input
        // first person
        System.out.println("First person data:");
        System.out.print("Name: ");
        name1 = sc.nextLine();
        System.out.print("Age: ");
        age1 = sc.nextInt();
        System.out.println();

        // second person
        System.out.println("Second person data:");
        System.out.print("Name: ");
        sc.nextLine();
        name2 = sc.nextLine();
        System.out.print("Age: ");
        age2 = sc.nextInt();
        System.out.println();

        // calculations
        average = (double)(age1 + age2) / 2;

        // answers
        System.out.print("-> The average age between " + name1 + " and " + name2 
        + " is " + String.format("%.2f", average) + ".");

        sc.close();
    }
}

File: cpp_problems/ages.cpp
#include <iostream>
#include <iomanip>
#include <climits>

using namespace std;

int main()
{
    // variables
    string name1, name2;
    int age1, age2;
    double average;

    // inputs
    cout << "First person data:" << endl;
    cout << "Name: ";
    getline(cin, name1);
    cout << "Age: ";
    cin >> age1;
    cout << endl;
    cout << "Second person data:" << endl;
    cout << "Name: ";
    cin.ignore(INT_MAX, '\n');
    getline(cin, name2);
    cout << "Age: ";
    cin >> age2;
    cout << endl;

    // calculation
    average = (double)(age1 + age2) / 2;

    // results
    cout << fixed << setprecision(1);
    cout << "The average age between " << name1 << " and " << name2 << " is " << average << ".";

    return 0;
}